distro/exporter.go

```diff
@@ -74,18 +74,40 @@ func newOTLPTracesExporter(_ logr.Logger, c *exporterConfig) (trace.SpanExporter
 		)
 	}
 
+	headers := make(map[string]string)
+	if c.AccessToken != "" {
+		headers["X-Sf-Token"] = c.AccessToken
+	}
+	isNone := noneEnvVarSet(otelExporterOTLPEndpointKey, otelExporterOTLPTracesEndpointKey, splunkRealmKey)
+	protocol := otlpProtocol(l, otelTracesExporterOTLPProtocolKey)
+
+	if protocol == otlpProtocolHTTPProtobuf {
+		var opts []otlptracehttp.Option
+
+		if len(headers) > 0 {
+			opts = append(opts, otlptracehttp.WithHeaders(headers))
+		}
+
+		if c.TLSConfig != nil {
+			opts = append(opts, otlptracehttp.WithTLSClientConfig(c.TLSConfig))
+		} else if isNone {
+			// Assume that the default endpoint (local collector) is non-TLS.
+			opts = append(opts, otlptracehttp.WithInsecure())
+		}
+
+		return otlptracehttp.New(ctx, opts...)
+	}
```

Comment:
This call to `otlpProtocol` is unnecessary if `isNone` is `true` right?

```suggestion
	if !isNone {
		protocol := otlpProtocol(l, otelTracesExporterOTLPProtocolKey)

		if protocol == otlpProtocolHTTPProtobuf {
			var opts []otlptracehttp.Option

			if len(headers) > 0 {
				opts = append(opts, otlptracehttp.WithHeaders(headers))
			}

			if c.TLSConfig != nil {
				opts = append(opts, otlptracehttp.WithTLSClientConfig(c.TLSConfig))
			} else if isNone {
				// Assume that the default endpoint (local collector) is non-TLS.
				opts = append(opts, otlptracehttp.WithInsecure())
			}

			return otlptracehttp.New(ctx, opts...)
		}
	}
```

Reply:
`isNone` is used to check whether the connection should be insecure. I’ve renamed the variable to avoid confusion. The same check was also present in the gRPC example, so I decided to extract it in order to reduce duplicated code.

Renamed in a74aeb9

Reply:
> This call to otlpProtocol is unnecessary if isNone is true right?

I do not think so.
`isNone` refers to endpoint URL and `otlpProtocol` to the protocol being used.
When `isNone` is `true` then still we need to check what is `otlpProtocol` to check whether we should use `http/protobuf` or `grpc` when sending telemetry to the local collector.